packages/platform-core/src/errors/amplify_error.ts

```diff
@@ -98,11 +98,14 @@ export abstract class AmplifyError<T extends string = string> extends Error {
     return undefined;
   };
 
-  static fromError = (error: unknown): AmplifyError<'UnknownFault'> => {
+  static fromError = (
+    error: unknown,
+    message?: string
+  ): AmplifyError<'UnknownFault'> => {
```

Comment:
This is confusing API, e.g. if I pass `message`, the implementation wouldn't use it if the `error` is instance of `Error` which is counter intuitive.

The client of this should pass `new Error(message)` if there is a message that should take precedence over `error.message` or `error` is not an instance of `Error`